### PR TITLE
feat(alloy): add `"full"` feature flag

### LINKED ISSUE #876: [Feature] Add `"full"` feature flag to `alloy` meta-crate
### Component

alloy

### Describe the feature you would like

Should:

- Include all the core re-exports
- Features: 

```
full = [
    "consensus",
    "contract",
    "eips",
    "k256",
    "kzg",
    "network",
    "provider-http", # includes `providers`
    "provider-ws",   # includes `providers`
    "provider-ipc",  # includes `providers`
    "rpc-types",     # includes `rpc-types-eth`
    "signer-wallet", # includes `signers`
]
```

### Additional context

cc @gakonst

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Closes #876 

## Solution

- Makes sure to enable `rpc-client` when enabling `rpc-client-ws` so that you do not also need to add the `rpc-client` flag if you if `rpc-client-ws`. This behavior is in line with the other feature flags.
- For easy of access make select `rpc-client` features (`WsConnect`) and (`IpcConnect`) directly available (if defined as flag) on the provider. Previously this required separately defining the `rpc-client` flag and importing it from the `rpc::client` namespace.

Related PR (see impact of proposed changes): https://github.com/alloy-rs/examples/pull/106

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/alloy/Cargo.toml [hunk 1]
```diff
@@ -72,6 +72,21 @@ std = [
...
+    "eips",
+    "k256",
+    "kzg",
+    "network",
+    "provider-http", # includes `providers`
+    "provider-ws",   # includes `providers`
+    "provider-ipc",  # includes `providers`
+    "rpc-types",     # includes `rpc-types-eth`
+    "signer-wallet", # includes `signers`
+]
```
Comment (zerosnacks) [inline crates/alloy/Cargo.toml:88]: In `examples` we did not need any specific flags from `core` as all commonly used functionality is already available by default

**Comment by zerosnacks** [comment]
Moving to review, open to feedback / suggestions

**Review by gakonst — APPROVED** [review]
Tested it out and I like this as a set of minimal features. I wonder if we could trim KZG, but seems fine for now.

File: crates/alloy/Cargo.toml [hunk 2]
```diff
@@ -72,6 +72,21 @@ std = [
     "alloy-consensus?/std",
 ]
 
+# full
+full = [
+    "consensus",
+    "contract",
+    "eips",
+    "k256",
+    "kzg",
```
Comment (mattsse) [inline crates/alloy/Cargo.toml:81]: imo this makes sense to include in `full`

full will probably include more stuff than necessary but for convenience this is okay imo.

**Review by mattsse — APPROVED** [review]

**Pull request merged**